File: 백준/Silver/2502. 떡 먹는 호랑이/떡 먹는 호랑이.js
const input = require("fs")
  .readFileSync("/dev/stdin", "utf8")
  .trim()
  .split("\n")
  .map((line) => line.replace(/\r/, ""));


const solution = (input) => {
  const [endDay, total] = input[0].split(" ").map((v) => Number(v));

  for (let i = 1; i <= total; i++) {
    for (let j = i; j <= total; j++) {
      const dp = Array(endDay).fill(0);
      dp[0] = i;
      dp[1] = j;
      for (let i = 2; i < endDay; i++) {
        dp[i] = dp[i - 1] + dp[i - 2];
      }

      if (dp[endDay - 1] === total) {
        console.log(i);
        console.log(j);
        return;
      }
    }
  }
};

solution(input);

File: 백준/Silver/2502. 떡 먹는 호랑이/떡 먹는 호랑이.py
endDay, total = list(map(int, input().split()))
isFind = False
for i in range(1, total+1):
  if isFind:
    break
  for j in range(i, total+1):
    dp = [0]*endDay
    dp[0] = i
    dp[1] = j

    for idx in range(2, endDay):
      dp[idx] = dp[idx-1] + dp[idx-2]
    
    if dp[endDay-1] == total:
      print(i)
      print(j)
      isFind = True
      break
